LayoutKit/LayoutArrangement.swift

```diff
@@ -82,21 +116,46 @@ public struct LayoutArrangement {
     }
 
     /// Returns the views for the layout and all of its sublayouts.
-    private func makeSubviews() -> [UIView] {
+    private func makeSubviews(from recycler: ViewRecycler, prepareAnimation: Bool) -> [UIView] {
         let subviews = sublayouts.flatMap({ (sublayout: LayoutArrangement) -> [UIView] in
-            return sublayout.makeSubviews()
+            return sublayout.makeSubviews(from: recycler, prepareAnimation: prepareAnimation)
         })
-        if let view = layout.makeView() {
-            view.frame = frame
+        // If we are preparing an animation, then we don't want to update frames or configure views.
+        if let view = layout.makeView(from: recycler, configure: !prepareAnimation) {
+            if !prepareAnimation {
```

Comment:
Looks like there are conceptually two tree walks here, one to create or update the sub/superview tree and another to set the frames and other configured properties. Sometimes we want to run them together and sometimes separately. Perhaps this logic would be cleaner as separate functions instead of passing down the flag throughout?